### PR TITLE
refactor(react-pi): use the shared assistant-stream SSE decoder

### PR DESCRIPTION
**Pull request by okisdev**
## summary

- react-pi's client hand-rolled a full SSE line parser (field/colon split, comment heartbeats, CRLF/CR handling, multi-line data); it now wraps assistant-stream's SSEEventDecoder, the same decoder react-a2a and react-google-adk already use
- `createSseDecoder` keeps its exported shape (SseFrame frames), so `openPiEventStream` and every consumer are unchanged; pi-only concerns stay local (ping filtering, payload and thread validation, snapshot-first reconnect, cancellation)
- adds the first-party `assistant-stream@^0.3.39` dependency (same specifier react-a2a carries) with the lockfile regenerated

### already verified

- [x] full react-pi vitest suite: 13 files, 216 tests green, including the decoder edge-case fixtures (CRLF splits, comment lines, multi-line data)

### reviewer should verify

- [ ] a live pi event stream still decodes and reconnects (snapshot replaces state after a dropped connection)

<a href="https://cubic.dev/pr/assistant-ui/assistant-ui/pull/6341?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

<a href="https://rupic.dev/s/s2.kdBaYxGKZ_jA7XDiykzG6I3EsBkcAWF7jAlNtObLBfeLvKj8JDWZNT77vd4?ref=github" target="_blank" rel="noopener noreferrer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://rupic.dev/badges/track-dark-v1.svg">Track in Rupic</picture></a>

### CODE REVIEW

File: packages/react-pi/src/client/eventSource.ts [hunk 1]
```diff
@@ -385,29 +339,14 @@ export const openPiEventStream = (
             const { value, done } = result;
             if (done) {
               shouldCancel = false;
+              for (const frame of sseDecoder.push(textDecoder.decode())) {
+                handleFrame(frame);
+              }
```
[the remark this answers is not in the document]
Comment (okisdev) [reply packages/react-pi/src/client/eventSource.ts]: removed the flush entirely: the pre-refactor parser also dropped an unterminated trailing frame at EOF (its done branch was a bare break), so the dead decode() call was the only delta and parity is now exact. the loop also decodes through createSseDecoder now, so the tested wrapper is the production path.
